docs/src/pages/components/trap-focus/trap-focus.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Trap Focus React component
+components: Unstable_TrapFocus
+---
+
+# Trap Focus
+
+<p class="description">Trap focus within a DOM node.</p>
+
+> ⚠️ The component is experimental and unstable.
+
+`TrapFocus` is a utility component that manages focus for its descendants.
+This is useful when implementing overlays such as modal dialogs, which should not allow focus to escape while open.
+
+When `open={true}` the trap is enabled, and pressing <kbd>Tab</kbd> or <kbd>Shift+Tab</kbd> will rotate focus within the inner focusable elements of the component.
+
+- 📦 [1.5 kB gzipped](https://material-ui.com/size-snapshot).
+- ⚛️ Support portals
+
+## Example
+
+{{"demo": "pages/components/trap-focus/BasicTrapFocus.js"}}
+
+## Disable enforce focus
+
+Clicks within the focus trap behave normally; but clicks outside the focus trap are blocked.
+
+You can disable this behavior with the `disableEnforceFocus` prop.
+
+{{"demo": "pages/components/trap-focus/DisableEnforceFocus.js"}}
+
+## Disable restore focus
+
+The component restores the focus back to the previously focused element when it deactivates, for example back to the button which opened a dialog.
+
+You can disable this behavior with the `disableRestoreFocus` prop.
```

Comment:
We should either explain a concrete use case or not document it. Why would I want to disable it? What are the drawbacks of this prop? etc.

Reply:
I have removed it